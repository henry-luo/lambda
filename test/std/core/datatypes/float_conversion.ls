// Test: Floating-Point Type Conversion
// Category: core/datatypes
// Type: positive
// Expected: 1

// String to float conversion
let from_string = "3.14" as float
from_string
let from_int = 42 as float
from_int
let from_bool_true = true as float
from_bool_true
let from_bool_false = false as float
from_bool_false
let pi = 3.14159
pi
let to_string = pi as string
to_string
let to_int = pi as int
to_int
let to_bool1 = 1.0 as bool
to_bool1
let to_bool0 = 0.0 as bool
to_bool0
let nan = 0.0 / 0.0
nan
let inf = 1.0 / 0.0
inf
let nan_str = nan as string
nan_str
let inf_str = inf as string
inf_str
let scientific = "1.23e-4" as float
scientific
let hex_float = "0x1.fffffep+127" as float
hex_float
1
