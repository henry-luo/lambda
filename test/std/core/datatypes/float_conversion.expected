3.14
42
1.0
0.0
3.14159
"3.14159"
3
true
false
NaN
Infinity
"NaN"
"Infinity"
0.000123
3.40282e+38
